Add upgrade release note for port 8443 (#3180)
theforeman/foreman-documentation: guides/doc-Release_Notes/topics/katello.adoc

[id="katello-upgrade-warnings"]
== Upgrade Warnings

> #There are no upgrade warnings with Katello {KatelloVersion}.

Reviewer: ```suggestion
// There are no upgrade warnings with Katello {KatelloVersion}.
```

comments in asciidoc: https://docs.asciidoctor.org/asciidoc/latest/comments/#comment-lines

=== Port 8443 disabled by default

On {SmartProxyServer}, the deprecated port 8443 has been disabled by default on new installations and upgrades.

> Clients are expected to be using port 443 when communicating through a {SmartProxyServer}.

Reviewer 2: ```suggestion
Clients should be using port 443 when communicating through a {SmartProxyServer}.
```

Reviewer 3: We should avoid the use of "should" per IBM SG. How about:
```suggestion
Clients now rely on port 443 when communicating through a {SmartProxyServer}.
```

> If there are older clients still using port 8443, they should be re-configured either by using remote execution, the katello-ca-consumer RPM, or updating `/etc/rhsm/rhsm.conf` using `subscription-manager --port 443`.

Reviewer 3: ```suggestion
If there are older clients still using port 8443, re-configured them either by using remote execution, the katello-ca-consumer RPM, or using `subscription-manager --port 443` to update `/etc/rhsm/rhsm.conf`.
```

> For users that temporarily need access to port 8443 again, the installer can be executed with `{foreman-installer} --foreman-proxy-content-reverse-proxy true`.

Reviewer 3: ```suggestion
If you need temporary access to port 8443 again, you can execute `{foreman-installer} --foreman-proxy-content-reverse-proxy true`.
```

> This should only be used as a stop gap as port 8443 will be removed permanently in the next release.

Reviewer 3: ```suggestion
This should only be used as a stop gap as port 8443 will be removed permanently in the next release.
```
What does "a stop gap" mean?

Author: Something that is temporary because the user has no other options in that moment. And that they need to address the core issue.

Reviewer 3: I'm not blocking the PR on this, but it would be preferred to avoid jargon and reword it.

Author: I cleaned it up.

> There are no upgrade warnings with Katello {KatelloVersion}.

Reviewer 4: ```suggestion
#There are no upgrade warnings with Katello {KatelloVersion}.
```

=== Port 8443 disabled by default

On {SmartProxyServer}, the deprecated port 8443 has been disabled by default on new installations and upgrades.
Clients are expected to be using port 443 when communicating through a {SmartProxyServer}.

> If there are older clients still using port 8443, they should be re-configured either by using remote execution, the katello-ca-consumer RPM, or updating `/etc/rhsm/rhsm.conf` using `subscription-manager --port 443`.

Reviewer 4: Do we have a specific documented procedure for this that we could link to? While the release notes aren't directly used in downstream, a procedure would and AFAIK this will also be relevant to downstream customers.

Author: We could link to the host management guide, perhaps a sub-section but there is no direct procedure for this.

Reviewer 4: Should we have that? If so, who would be responsible for writing it?

Author: This can vary by use-case, and what feature sets are used by a given customer. And https://community.theforeman.org/t/rfc-replace-pub-katello-rhsm-consumer-static-script-with-a-template/38494/21 is standardizing this to an endpoint for configuration. I think the documentation that comes with that change should suffice.
